### PR TITLE
Documentation (docstrings and site) updates

### PR DESCRIPTION
**Pull request by abhro**
Main changes:
- Add missing dependency
- Add missing import
- Add more doctests and update output
Please see the commit messages for more details! Thank you!

### CODE REVIEW

**Comment by rafaqz** [comment]
Hi, thanks for this! 

Could you revert the files where only whitespace changed? Just so we can see what the real changes are to review

**Comment by abhro** [comment]
No worries! I completely understand about keeping the PRs focused! If there's any drama, it's between me and my text editor because it keeps complaining about whitespace lol. I'll get to the revert as soon as I can. Thank you!!

**Comment by abhro** [comment]
Revert commit for c54bc6c pushed! Thank you!

File: docs/src/broadcast_dims.md [hunk 1]
```diff
@@ -12,8 +12,13 @@ As an added bonus, `broadcast_dims` even works on `DimStack`s.
 
 Define some dimensions:
 
+````@example bd
```
Comment (rafaqz) [inline docs/src/broadcast_dims.md:15]: Whats the reason to split these examples?

**Review by rafaqz — APPROVED** [review]
Thanks so much, this is an extremely comprehensive docs edit. I can't believe there were so many typos.

**Comment by abhro** [comment]
Hi! Sorry for the delay, the comments had been sitting in "pending" for a bit, and I assumed that it was pending to get approved by a moderator.

File: docs/src/broadcast_dims.md (hunk 1 above)
Comment (rafaqz) [reply docs/src/broadcast_dims.md:15]: Makes sense to me!

**Comment by lazarusA** [comment]
it looks like some of the doctests are failing now. Did the expected output changed? then it will need to be modified. Or is use an issue of missing blank spaces?

**Comment by rafaqz** [comment]
You should be able to run the doctests again with `fix=true` to fix it

**Comment by abhro** [comment]
It is newline related, and Documenter.jl is a bit finicky in how it separates input/output for REPL doctests. I'll be pushing a commit soon to fix it. Thanks for bringing it to my attention!

**Review by rafaqz — APPROVED** [review]

**Comment by rafaqz** [comment]
Thanks again!

**Pull request merged**